app/src/main/java/com/example/parkingsystemjava/database/ReservationDataBase.java

```diff
@@ -34,11 +34,18 @@ public void addValues(Reservation reservation) {
 
     public Reservation getReservation(String parkingLots, String pass) {
         List<Reservation> listReservation = reservationHashMap.get(parkingLots);
-        for (Reservation reservation : listReservation) {
-            if (reservation.getUserPassword().equals(pass)) {
-                return reservation;
+        if (listReservation != null) {
+            for (Reservation reservation : listReservation) {
+                if (reservation.getUserPassword().equals(pass)) {
+                    return reservation;
+                }
             }
         }
         return null;
     }
+
+    public List<Reservation> getReservation(String parkingLots) {
+        List<Reservation> listReservation = reservationHashMap.get(parkingLots);
+        return listReservation;
```

Comment:
unify these lines please. try to not use extra variables

Reply:
Done